games/static/games/proto.js

```diff
@@ -0,0 +1,97 @@
+class Square{
+    constructor(indexOnBoard, div) {
+        this.index = indexOnBoard
+        this.div = div
+    }
+    setCharacter(character) {
+        this.character = character
+        this.div.class = character.getCssClass
+    }
+    isTaken() {}    
+}
+
+function createSquare(indexOnBoard, imageCssClass) {
+    let square = new Square(indexOnBoard, imageCssClass)
+    return square
+}
+
+class Defender{
+    constructor(imageCssClass, indexOnBoard, lineWidth) {
+        this.imageCssClass = imageCssClass
+        this.indexOnBoard = indexOnBoard
+        this.lineWidth = lineWidth
+    }
+}
+
+class Alien{
+    constructor(name, imageCssClass, health, indexOnBoard, lineWidth ) {
+        this.name = name
+        this.imageCssClass = imageCssClass
+        this.health = health
+        this.colorByHealthLevel = {
+            1: 'green',
+            2: 'blue'
+        }
+        this.indexOnBoard = indexOnBoard 
+        this.lineWidth = lineWidth
+    }
+    _move(direction) {
+        this.indexOnBoard = indexOnBoard + direction
+    }
+    moveLeft() {
+        this._move(-1)
+    } 
+    moveRight() {
+        this._move(1)
+    } 
+    moveDown() {
+        this._move(this.lineWidth)
+    }
+    isDead() {
+        return this.health === 0
+    }
+    draw(square) {
```

Comment:
if square is an instance of the Square class so we dont really need draw
we can do something like alien.moveTo(square):
```
moveTo(square) {
   square.SetCharecter(this)
}
getCssClass() {
       if(this.isDead()) {
            return 'empty'
        } 
        return this.imageCssClass 
}
```